src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -74,17 +74,25 @@ internal sealed partial class ServiceHubRemoteHostClient : RemoteHostClient
                 return;
             }
 
-            // RegisterWorkspaceHost is required to be called from UI thread so push the code
-            // to UI thread to run. 
-            await Task.Factory.SafeStartNew(() =>
+            // Create a connection to the host in the BG to avoid taking the hit of loading 
+            // service hub on the UI thread.
+            using (var connection = await client.TryCreateConnectionAsync(WellKnownRemoteHostServices.RemoteHostService, CancellationToken.None).ConfigureAwait(false))
             {
-                var projectTracker = vsWorkspace.GetProjectTrackerAndInitializeIfNecessary(Shell.ServiceProvider.GlobalProvider);
-
                 var host = new WorkspaceHost(vsWorkspace, client);
 
-                projectTracker.RegisterWorkspaceHost(host);
-                projectTracker.StartSendingEventsToWorkspaceHost(host);
-            }, CancellationToken.None, ForegroundThreadAffinitizedObject.CurrentForegroundThreadData.TaskScheduler).ConfigureAwait(false);
+                // RegisterWorkspaceHost is required to be called from UI thread so push the code
+                // to UI thread to run. 
+                await Task.Factory.SafeStartNew(() =>
+                {
+                    host.SetPreferredConnection(connection);
+                    var projectTracker = vsWorkspace.GetProjectTrackerAndInitializeIfNecessary(Shell.ServiceProvider.GlobalProvider);
+
+                    projectTracker.RegisterWorkspaceHost(host);
+                    projectTracker.StartSendingEventsToWorkspaceHost(host);
+
+                    host.SetPreferredConnection(null);
```

Comment:
> making it possible for unrelated code to use the preferred connection where it isn't expected to.

That's actually the point of this change :)

Right now we create the workspace host and then register it.  The project tracker then sends the host all the state changes it has tracked.  If any of these state changes need to be remoted to the OOP server, we have to expensively spin up a connection, and that impacts hte UI thread.  The point is to have the connection ready to go for this initial set of events to be sent over.  Once that happens, in teh future, we'll just acquire a connection on demand like we were doing before.